File: js/examples/test-wallet-balance-endpoint.js
require('dotenv').config();
const OptikpiDataPipelineSDK = require('../src/index');
const { WalletBalanceEvent } = require('../src/models');

// Configuration - Read from environment variables
const API_BASE_URL = process.env.API_BASE_URL;
const AUTH_TOKEN = process.env.AUTH_TOKEN;
const ACCOUNT_ID = process.env.ACCOUNT_ID;
const WORKSPACE_ID = process.env.WORKSPACE_ID;

// Validate required environment variables
if (!AUTH_TOKEN || !ACCOUNT_ID || !WORKSPACE_ID) {
  console.error('❌ Error: Missing required environment variables!');
  console.error('   Please set: AUTH_TOKEN, ACCOUNT_ID, WORKSPACE_ID');
  console.error('   Copy env.example to .env and fill in your values');
  process.exit(1);
}

// Initialize SDK
const sdk = new OptikpiDataPipelineSDK({
  authToken: AUTH_TOKEN,
  accountId: ACCOUNT_ID,
  workspaceId: WORKSPACE_ID,
  baseURL: API_BASE_URL
});

// Wallet Balance event data
const walletBalance =new WalletBalanceEvent({
  "account_id": ACCOUNT_ID,
  "workspace_id": WORKSPACE_ID,
  "user_id": "user123456",
  "event_category": "Wallet Balance",
  "event_name": "Balance Update",
  "event_id": "evt_wb_987654321",
  "event_time": "2024-01-15T14:45:00Z",
  "wallet_type": "main",
  "currency": "USD",
  "current_cash_balance": 1250.50,
  "current_bonus_balance": 100.00,
  "current_total_balance": 1350.50,
  "blocked_amount": 50.00
});

// Test wallet balance endpoint
async function testWalletBalanceEndpoint() {
  try {
    console.log('🚀 Testing Wallet Balance Events Endpoint');
    console.log('==========================================');
    
    console.log('Configuration:');
    console.log(`API Base URL: ${API_BASE_URL}`);
    console.log(`Account ID: ${ACCOUNT_ID}`);
    console.log(`Workspace ID: ${WORKSPACE_ID}`);
    console.log(`Auth Token: ${AUTH_TOKEN.substring(0, 8)}...`);
    
    console.log('\nMaking API request using SDK...');
    console.log('Wallet Balance Event Data:', JSON.stringify(walletBalance, null, 2));
    
    // Make the API call using SDK
    const startTime = Date.now();
    const result = await sdk.sendWalletBalanceEvent(walletBalance);
    const endTime = Date.now();
    
    if (result.success) {
      console.log('\n✅ Success!');
      console.log('==========================================');
      console.log(`HTTP Status: ${result.status}`);
      console.log(`Response Time: ${endTime - startTime}ms`);
      console.log(`Success: ${result.success}`);
      console.log('Response Data:', JSON.stringify(result.data, null, 2));
    } else {
      console.log('\n❌ API Error!');
      console.log('==========================================');
      console.log(`HTTP Status: ${result.status}`);
      console.log(`Response Time: ${endTime - startTime}ms`);
      console.log(`Success: ${result.success}`);
      console.log('Error Data:', JSON.stringify(result.data, null, 2));
    }
    
  } catch (error) {
    console.error('\n❌ Error occurred!');
    console.error('==========================================');
    if (error.response) {
      console.error(`HTTP Status: ${error.response.status}`);
      console.error('Error Response:', JSON.stringify(error.response.data, null, 2));
    } else {
      console.error('Error:', error.message);
      console.error('Stack:', error.stack);
    }
  }
}

// Run the test
if (require.main === module) {
  testWalletBalanceEndpoint();
}

module.exports = {
  testWalletBalanceEndpoint,
  walletBalance,
  sdk
};

File: python/examples/test-wallet-balance-endpoint.py
#!/usr/bin/env python3
"""
Test Wallet Balance Endpoint
Demonstrates how to send wallet balance event data to the Optikpi Data Pipeline API
"""
import os
import sys
import json
import time
from pathlib import Path
from dotenv import load_dotenv

# Add parent directory to path
sys.path.insert(0, str(Path(__file__).parent.parent / "src" / "python"))

from index import OptikpiDataPipelineSDK
from models.WalletBalanceEvent import WalletBalanceEvent

# Load environment variables
load_dotenv()

# Configuration - Read from environment variables
API_BASE_URL = os.getenv("API_BASE_URL")
AUTH_TOKEN = os.getenv("AUTH_TOKEN")
ACCOUNT_ID = os.getenv("ACCOUNT_ID")
WORKSPACE_ID = os.getenv("WORKSPACE_ID")

# Validate required environment variables
if not AUTH_TOKEN or not ACCOUNT_ID or not WORKSPACE_ID:
    print('❌ Error: Missing required environment variables!', file=sys.stderr)
    print('   Please set: AUTH_TOKEN, ACCOUNT_ID, WORKSPACE_ID', file=sys.stderr)
    print('   Copy env.example to .env and fill in your values', file=sys.stderr)
    sys.exit(1)

# Initialize SDK
sdk = OptikpiDataPipelineSDK({
    'authToken': AUTH_TOKEN,
    'accountId': ACCOUNT_ID,
    'workspaceId': WORKSPACE_ID,
    'baseURL': API_BASE_URL
})

# Wallet balance event data
wallet = WalletBalanceEvent(
    account_id=ACCOUNT_ID,
    workspace_id=WORKSPACE_ID,
    user_id="1wallet_py_1",
    event_category="Wallet Balance",
    event_name="Balance Updated",
    event_id="evt_wb_1234",
    event_time="2024-01-15T11:30:00Z",
    currency="USD",
    current_cash_balance=500.00,
    current_bonus_balance=50.00,
    current_total_balance=550.00,
    blocked_amount=20.00
)

validation = wallet.validate()

if not validation.get("isValid", False):
    print("❌ Validation errors:", validation.get("errors", []))
    sys.exit(1)

print("✅ Wallet balance event validated successfully!")


def test_wallet_balance_endpoint():
    """
    Test wallet balance endpoint
    
    Sends wallet balance event data and displays the response
    """
    try:
        print('🚀 Testing Wallet Balance Events Endpoint')
        print('========================================')

        print('Configuration:')
        print(f'API Base URL: {API_BASE_URL}')
        print(f'Account ID: {ACCOUNT_ID}')
        print(f'Workspace ID: {WORKSPACE_ID}')
        print(f'Auth Token: {AUTH_TOKEN[:8]}...')

        print('\nMaking API request using SDK...')
        wallet_dict = wallet.to_dict()
        print(f'Wallet Event Data: {json.dumps(wallet_dict, indent=2)}')

        start_time = time.time()
        result = sdk.send_wallet_balance_event(wallet_dict)
        end_time = time.time()

        response_time = int((end_time - start_time) * 1000)

        if result['success']:
            print('\n✅ Success!')
            print('========================================')
            print(f"HTTP Status: {result['status']}")
            print(f"Response Time: {response_time}ms")
            print(f"SDK Success: {result['success']}")
            print(f"Response Data: {json.dumps(result['data'], indent=2)}")
        else:
            print('\n❌ API Error!')
            print('========================================')
            print(f"HTTP Status: {result['status']}")
            print(f"Response Time: {response_time}ms")
            print(f"SDK Success: {result['success']}")
            print(f"Error Data: {json.dumps(result.get('data', result.get('error')), indent=2)}")

    except Exception as error:
        print('\n❌ SDK Error occurred!', file=sys.stderr)
        print('========================================', file=sys.stderr)
        print(f'Error: {str(error)}', file=sys.stderr)
        import traceback
        print(f'Stack: {traceback.format_exc()}', file=sys.stderr)


# Run the test
if __name__ == "__main__":
    test_wallet_balance_endpoint()
